tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -0,0 +1,676 @@
+        }
+    }
+
+    private unsafe void GenerateCrashReport(int pid, int? signal, int? crashThread)
+    {
+        var lib = NativeLibrary.Load(Path.Combine(AppContext.BaseDirectory, "Datadog.Profiler.Native.so"));
+
+        var export = NativeLibrary.GetExport(lib, "CreateCrashReport");
+
+        if (export == 0)
+        {
+            Errors.Add("Failed to load the CreateCrashReport function");
+            return;
+        }
+
+        using var target = DataTarget.AttachToProcess(pid, suspend: true);
+        _runtime = target.ClrVersions[0].CreateRuntime();
+
+        var function = (delegate* unmanaged<int, IntPtr>)export;
+
+        var ptr = function(pid);
+
+        if (ptr == IntPtr.Zero)
+        {
+            Errors.Add("Failed to create crash report");
+            return;
+        }
+
+        using var iunknown = NativeObjects.IUnknown.Wrap(ptr);
+
+        int result = iunknown.QueryInterface(ICrashReport.Guid, out var crashReportPtr);
+
+        if (result != 0)
+        {
+            Errors.Add($"Failed to query interface: {result}");
+            return;
+        }
+
+        using var crashReport = NativeObjects.ICrashReport.Wrap(crashReportPtr);
+
+        try
+        {
+            crashReport.Initialize();
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to initialize crash report: {GetLastError(crashReport, ex)}");
+            return;
+        }
+
+        if (crashThread == null)
+        {
+            var firstThreadWithException = _runtime.Threads.FirstOrDefault(t => t.CurrentException != null);
+
+            if (firstThreadWithException != null)
+            {
+                crashThread = (int)firstThreadWithException.OSThreadId;
+            }
+        }
+
+        ClrException? exception = null;
+
+        // Check if there's an exception on the crash thread
+        if (crashThread != null)
+        {
+            exception = _runtime.Threads.FirstOrDefault(t => t.OSThreadId == crashThread.Value)?.CurrentException;
+        }
+
+        bool isSuspicious;
+
+        try
+        {
+            var callback = (delegate* unmanaged<IntPtr, ResolveMethodData*, int>)&ResolveManagedMethod;
+            crashReport.ResolveStacks(crashThread ?? 0, (IntPtr)callback, out isSuspicious);
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to resolve stacks: {GetLastError(crashReport, ex)}");
+            return;
+        }
+
+        if (!isSuspicious && exception != null)
+        {
+            // The stacks aren't suspicious, but maybe the exception is
+            var exceptionType = exception.Type?.Name ?? string.Empty;
+
+            var suspiciousExceptionTypes = new[] { "System.InvalidProgramException", "System.Security.VerificationException", "System.MissingMethodException", "System.BadImageFormatException" };
+
+            if (exceptionType.StartsWith("Datadog") || suspiciousExceptionTypes.Contains(exceptionType))
```

Comment:
As before, should we look for `Datadog`? Also probably should have either `StringComparison.Ordinal` or `StringComparison.OrdinalIgnoreCase`?

Reply:
The case is unlikely to be wrong, but I guess it can't hurt.

I prefer to look for `Datadog` rather than `Datadog.`, just in case